scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+
+    Also See
+    --------
+    single_area_plot
+
+    """
+
+    # Parses the taxonomy and collapses the table
+    taxa = extract_label_array(table, tax_col, tax_delim)
+    if samples is None:
+    	samples = list(table.drop(columns=[tax_col]).columns)
+
+    # Gets the appropriate taxonomic level information to go forward
+    collapsed = level_taxonomy(table, taxa, samples, 
+                               level=np.array([fine_level]), 
+                               consider_nan=multilevel_table)
+    samples = collapsed.columns
+
+    # Gets the top taxonomic levels
+    upper_, lower_, = profile_joint_levels(collapsed, rough_level, fine_level, 
+    								       samples=samples,
+                                           lo_thresh=abund_thresh_rough, 
+                                           lo_count=min(5, group_thresh_rough),
+                                           hi_thresh=abund_thresh_fine,
+                                           hi_count=group_thresh_fine,
+                                           )
+    # Gets the colormap 
+    cmap = define_join_cmap(upper_)
+    upper_.index = upper_.index.droplevel('rough')
+    lower_.index = lower_.index.droplevel('rough')
+
+    # Plots the data
+    fig_ = plot_area(upper_.astype(float), lower_.astype(float), cmap)
+
+    return fig_
+
+
+# Sets up the main arguments for argparse.
+parser_one = argparse.ArgumentParser(
+    description=('A set of functions to generate diagnostic stacked area '
+                 'plots from metagenomic outputs.'),
+    prog=('area_plotter'),
+    )
+parser_one.add_argument('-t', '--table', 
+                    help=('The abundance table as a tsv classic biom '
+                          '(features as rows, samples as columns) containing'
+                          'absloute or relative abundance for the samples.'),
+                    required=True,
+                    )
+parser_one.add_argument('-f', '--figure',
+                    help=('The location for the final figure'),
+                    required=True,
+                    )
+parser_one.add_argument('-s', '--samples', 
+                    help=('A text file with the list of samples to be include' 
+                         ' (one per line). If no list is provided, then data '
+                         'from all columns in the table (except the one '
+                         'specifying taxonomy) will be used.'),
+                    )
+parser_one.add_argument('--mode', choices=(["metaphlan", "kracken", "marker"]),
+                    help=('The software generating the table to make parsing'
+                          'easier. Optens are kracken, metaphlan, marker '
+                          '(i.e. CTMR amplicon) and other for all other '
+                          'processing types.'),
+                    default='kracken'
+                    )
+parser_one.add_argument('-l', '--level',
+                    help=('The taxonomic level (as an integer) to plot the '
```

Comment:
Please include the default values for **all** parameters in the help text, so users can easily see them, e.g.:

```
help=('The taxonomic level used to plot the data (integer in the range 1-7) [%(default)s]. '
```

This will be formatted like this in the help output:

```
help=('The taxonomic level used to plot the data (integer in the range 1-7) [3]. '
```

Reply:
For this specific argument, it could be useful to specify it as a `choices` alternative as well, since it's a small range of valid integers:
`choices=range(1, 7)` or something would do it, right?

Reply:
We don't know, apriori, the number of possible levels which is why I stayed away from choices. For example, a strain-level table might have 8 levels, a gene might only have 4. This was part of the reason we didn't specify something like names and instead left integers.

Reply:
Good point!